src/Compilers/Test/Core/Diagnostics/DiagnosticExtensions.cs

```diff
@@ -362,7 +362,7 @@ public static TCompilation VerifyEmitDiagnostics<TCompilation>(this TCompilation
             IEnumerable<ResourceDescription> manifestResources = null)
             where TCompilation : Compilation
         {
-            var pdbStream = MonoHelpers.IsRunningOnMono() ? null : new MemoryStream();
+            var pdbStream = MonoHelpers.IsRunningOnMono() || options?.EmitMetadataOnly == true ? null : new MemoryStream();
```

Comment:
why do we need this change?

Reply:
Allows verifying emit diagnostics when we are only emitting a ref assembly.